fastly/custom_tls_configuration.go

```diff
@@ -0,0 +1,153 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// CustomTLSConfiguration represents a TLS configuration response from the Fastly API.
+type CustomTLSConfiguration struct {
+	ID            string       `jsonapi:"primary,tls_configuration"`
+	DNSRecords    []*DNSRecord `jsonapi:"relation,dns_records"`
+	Bulk          bool         `jsonapi:"attr,bulk"`
+	Default       bool         `jsonapi:"attr,default"`
+	HTTPProtocols []string     `jsonapi:"attr,http_protocols"`
+	Name          string       `jsonapi:"attr,name"`
+	TLSProtocols  []string     `jsonapi:"attr,tls_protocols"`
+	CreatedAt     *time.Time   `jsonapi:"attr,created_at,iso8601"`
+	UpdatedAt     *time.Time   `jsonapi:"attr,updated_at,iso8601"`
+}
+
+// DNSRecord is a child of CustomTLSConfiguration
+type DNSRecord struct {
+	ID         string `jsonapi:"primary,dns_record"`
+	RecordType string `jsonapi:"attr,record_type"`
+	Region     string `jsonapi:"attr,region"`
+}
+
+// ListCustomTLSConfigurationsInput is used as input to the ListCustomTLSConfigurationsInput function.
+type ListCustomTLSConfigurationsInput struct {
+	FilterBulk *bool   // Whether or not to only include bulk=true configurations
+	Include    *string // Include related objects. Optional, comma-separated values. Permitted values: dns_records.
+	PageNumber *uint   // The page index for pagination.
+	PageSize   *uint   // The number of keys per page.
+
+}
+
+// formatFilters converts user input into query parameters for filtering.
+func (i *ListCustomTLSConfigurationsInput) formatFilters() map[string]string {
+	result := map[string]string{}
+	pairings := map[string]interface{}{
+		"filter[bulk]": i.FilterBulk,
+		"include":      i.Include,
+		"page[size]":   i.PageSize,
+		"page[number]": i.PageNumber,
+	}
+	for key, value := range pairings {
+		if !reflect.ValueOf(value).IsNil() {
+			result[key] = fmt.Sprintf("%v", reflect.ValueOf(value).Elem())
+		}
+	}
+	return result
+}
+
+// ListCustomTLSConfigurations list all TLS configurations.
+func (c *Client) ListCustomTLSConfigurations(i *ListCustomTLSConfigurationsInput) ([]*CustomTLSConfiguration, error) {
+	p := "/tls/configurations"
+	filters := &RequestOptions{
```

Comment:
to be consistent with other files can we change from `filters` to `ro` (will require changes to references elsewhere in this file) also fits more closely with Go's idiom of naming variables after the associated type.

```suggestion
	ro := &RequestOptions{
```

Reply:
Updated